### ISSUE TITLE
Bad output from generator

### ISSUE THREAD

**Issue by jlumpe** [NVlabs/stylegan2-ada-pytorch#62]
**Describe the bug**

Values in output array are in the incorrect range, expect -1 to 1 but got approximately -150 - 50. When attempting to transform to the 0-255 range using the transformation `x * 127.5 + 128` all values are clipped at the extremes. Changing this to a scale of 1 and shift of 170 all values are put in the right range, but image is still full of artifacts. See screenshots below.

Both images do contain the shape of a face, so something is working correctly.

**To Reproduce**

Can get this by cloning the repo and running the example command:

```
python generate.py --outdir=out --trunc=1 --seeds=85,265,297,849 \
      --network=https://nvlabs-fi-cdn.nvidia.com/stylegan2-ada-pytorch/pretrained/metfaces.pkl
```

I used the following code based off of `generate.py` to rescale the image values:

```python
import pickle
import torch
from PIL import Image


DEVICE = 'cuda:0'

def transform_output(a, scale, offset):
    return (a.permute(0, 2, 3, 1) * scale + offset).clamp(0, 255).to(torch.uint8)

def latent_from_seed(seed):
    return torch.from_numpy(np.random.RandomState(seed).randn(1, G.z_dim)).to(DEVICE) 


# Load net
with open('pretrained/metfaces.pkl', 'rb') as f:
    data = pickle.load(f)

G = data['G_ema'].cuda()


# Generate
z = latent_from_seed(85)
a = G(z, None, truncation_psi=1, noise_mode='const')


# Image using standard transformation
a2 = transform_output(a, scale=127.5, offset=128)
img = Image.fromarray(a2[0].cpu().numpy(), 'RGB')


# Image using alternate transformation
a2 = transform_output(a, scale=1, offset=170)
img = Image.fromarray(a2[0].cpu().numpy(), 'RGB')
```

**Screenshots**

Output of `metfaces` network using seed 85 in example command:

![seed0085](https://user-images.githubusercontent.com/7455396/110372651-cf8c3500-800b-11eb-900a-33ead2bbb092.png)

After attempting to scale/shift values into the correct range:

![image](https://user-images.githubusercontent.com/7455396/110373238-838dc000-800c-11eb-882b-37921103033d.png)

Distribution of channel values in output of generator:

![image](https://user-images.githubusercontent.com/7455396/110373310-97d1bd00-800c-11eb-92c7-d80bb679973b.png)

**System info**

 - Pop!_OS 20.10 (Ubuntu 20.10)
 - pytorch 1.7.1
 - CUDA toolkit 11.2
 - NVIDIA driver 460.39
 - GTX 1650 4GB (figure this should be OK for just running the generator?)
 - No docker

**Comment by jeffshee** [comment]
Hi, I'm facing a similar problem as well. 

After some training, I inspected the RGB range of generated images and real images, and it turns out to be `(-2.6337, 3.3891)` and `(-1,1)`, respectively. It's very weird since the generated images are expected to be very close to real images at this stage, yet even a simple discriminator can easily discriminate between generated/real by only looking at the input's range in this case. 

After some investigation, I realized that they are not using `tanh` at the last layer of the generator. Is there anyone who could explain why? Someone explained it here: https://github.com/taki0112/StyleGAN-Tensorflow/issues/5 but it's not that convincing IMO.

@jlumpe Maybe try adding `tanh` as mentioned above?

**Comment by ericmoderbacher** [comment]
I'm getting similarly bad output from the generator using the pretrained values. With no other indication of errors. 

I'm able to use docker in my environment.  But I'm pretty sure I have everything set up correctly.

The tensorflow version of this project works without issues.

Sample output:
![seed0297](https://user-images.githubusercontent.com/8312116/127032700-f9b6f705-c43b-427e-a208-eca571220d6b.png)

**Comment by panegri** [comment]
> I'm getting similarly bad output from the generator using the pretrained values. With no other indication of errors.
> 
> I'm not able to use docker in my environment. But I'm pretty sure I have everything set up correctly.
> 
> The tensorflow version of this project works without issues.
> 
> Sample output:
> ![seed0297](https://user-images.githubusercontent.com/8312116/127032700-f9b6f705-c43b-427e-a208-eca571220d6b.png)

eric, I had this kind of output for Windows, but using Linux in a same environment, the output is an aceptable face. Any idea?

**Comment by ericmoderbacher** [comment]
@panegri Nope, no idea why.  I never solved that issue. 😞

**Comment by panegri** [comment]
Hi @ericmoderbacher , 
I think I find a workaround.
Using legacy.py, I converted the weights from tensorflow:
>python legacy.py --source=https://nvlabs-fi-cdn.nvidia.com/stylegan2/networks/stylegan2-ffhq-config-f.pkl --dest=models/ffhq2.pkl
Now the ffhq2.pkl generates a face correctly.
I hope it helps.
Pablo

**Comment by ericmoderbacher** [comment]
@panegri oh I didn't think of that, thank you for telling me about it!

**Comment by Yeats-CJ** [comment]
It seems like a Nvidia and Cuda problem, because it can generate great output when I change to use cpu. Recommend to use linux os and check the Nvidia or cuda setting.

**Comment by Timsword** [comment]
> Hi @ericmoderbacher , I think I find a workaround. Using legacy.py, I converted the weights from tensorflow:
> 
> > python legacy.py --source=https://nvlabs-fi-cdn.nvidia.com/stylegan2/networks/stylegan2-ffhq-config-f.pkl --dest=models/ffhq2.pkl
> > Now the ffhq2.pkl generates a face correctly.
> > I hope it helps.
> > Pablo

Hi, sorry i'm a little bit dark in here. How can i implement it inside the legacy.py? I was using the notebook below but i run the same issue.
https://github.com/jeffheaton/stylegan2-toys/blob/main/morph_video_real.ipynb
